Add indexof method in string/terms (#1857)
Codecademy/docs: content/c-sharp/concepts/strings/terms/indexof/indexof.md

---
Title: '.IndexOf()'
Description: 'Returns the index of a specified character or substring in a string.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Methods'
  - 'Strings'

>   - 'String Searching'

Reviewer: ```suggestion
  - 'Search'
```

Reviewer: 'string searching' isn't currently on our tag list

CatalogContent:
  - 'learn-c-sharp'
  - 'paths/computer-science'
---

> The **`IndexOf()`** method is a string method that returns the index of the first occurrence of a specified character or substring in a given string. If the character or substring is not found, it returns `-1`.

Reviewer: ```suggestion
The **`.IndexOf()`** method is a string method that returns the index of the first occurrence of a specified character or substring in a given string. If the character or substring is not found, it returns `-1`.
```

## Syntax

```pseudo

> public int IndexOf(char value);
> public int IndexOf(string value);
> public int IndexOf(char value, int startIndex);
> public int IndexOf(string value, int startIndex);
> public int IndexOf(string value, int startIndex, int count);

Reviewer: ```suggestion
IndexOf(char value);
IndexOf(string value);
IndexOf(char value, int startIndex);
IndexOf(string value, int startIndex);
IndexOf(string value, int startIndex, int count);
```

```

- `value` : The substring or character to search for.
- `startIndex` (optional): The index from where the search should start.
- `count` (optional): Maximum number of characters to search.

> ## Example

Reviewer: For simple methods one example is usually sufficient. I think the second example is more comprehensive and interesting, so please remove the first example.

The following example uses the `IndexOf()` method to find the index of the first occurrence of the character "d" in the string "Codecademy docs".

```cs
string str = "Codecademy docs";
int index = str.IndexOf('d');

Console.WriteLine("Index: " + index);
```

This example results in the following output:

```
Index: 2
```

## Codebyte Example

> The following codebyte is runnable and shows the usage of the `IndexOf()` method on a string.

Reviewer: ```suggestion
The following codebyte is runnable and shows the usage of the `.IndexOf()` method on a string.
```

```codebyte/csharp
using System;

public class IndexOfExample
{
    public static void Main()
    {
        string sentence = "The cat in the hat wears a red and white striped hat";
        
        Console.WriteLine(sentence);
        
        int index1 = sentence.IndexOf('h');
        Console.WriteLine($"Index of 'h': {index1}");
        
        int index2 = sentence.IndexOf("cat");
        Console.WriteLine($"Index of 'cat': {index2}");
        
        int index3 = sentence.IndexOf('a', 10);
        Console.WriteLine($"Index of 'a' starting from index 10: {index3}");
        
        int index4 = sentence.IndexOf("dog");
        Console.WriteLine($"Index of 'dog': {index4}");
    }
}
```

> This example results in the following output: 

Reviewer: We don't include a separate output for codebyte examples. You can remove the results below.

```shell
The cat in the hat wears a red and white striped hat
Index of 'h': 11
Index of 'cat': 4
Index of 'a' starting from index 10: 16
Index of 'dog': -1
```

> This example used the `.IndexOf()` method to find the index of the first occurrence of the letter `'h'`, the substring `"cat"`, the letter `'a'` starting from index `10`, and the substring `"dog"`. Since `"dog"` is not found in the string, the method returns `-1`.

Reviewer: ```suggestion
This example used the `.IndexOf()` method to find the index of the first occurrence of the letter `"h"`, the substring `"cat"`, the letter `"a"` starting from index `10`, and the substring `"dog"`. Since `"dog"` is not found in the string, the method returns `-1`.
```

> In this example We use the `IndexOf()` method to find the index of the first occurrence of the letter `'h'`, the substring `"cat"`, the letter `'a'` starting from index `10`, and the substring `"dog"`. Since `"dog"` is not found in the string, the method returns `-1`.

Reviewer: ```suggestion
This example used the `.IndexOf()` method to find the index of the first occurrence of the letter `'h'`, the substring `"cat"`, the letter `'a'` starting from index `10`, and the substring `"dog"`. Since `"dog"` is not found in the string, the method returns `-1`.
```

---
Title: '.IndexOf()'
Description: 'Returns the index of a specified character or substring in a string.'
Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Methods'
  - 'Strings'
  - 'String Searching'
CatalogContent:
  - 'learn-c-sharp'
  - 'paths/computer-science'
---

The **`IndexOf()`** method is a string method that returns the index of the first occurrence of a specified character or substring in a given string. If the character or substring is not found, it returns `-1`.

## Syntax

```pseudo
public int IndexOf(char value);
public int IndexOf(string value);
public int IndexOf(char value, int startIndex);
public int IndexOf(string value, int startIndex);
public int IndexOf(string value, int startIndex, int count);
```

> -  `value` : The substring or character to search for.

Reviewer 2: ```suggestion
- `value` : The substring or character to search for.
```

- `startIndex` (non-mandatory): The index from where the search should start.
- `count` (non-mandatory): Maximum number of characters to search.

> 

Reviewer 2: ```suggestion
```
Remove extra line break.

## Example

> The following example uses the `IndexOf()` method to find the index of the first occurrence of the character 'd' in the string "Codecademy docs".

Reviewer 2: ```suggestion
The following example uses the `IndexOf()` method to find the index of the first occurrence of the character "d" in the string "Codecademy docs".
```

```cs
string str = "Codecademy docs";
int index = str.IndexOf('d');

Console.WriteLine("Index: " + index);
```

This example results in the following output:

```
Index: 2
```

> 

Reviewer 2: ```suggestion
```
Remove extra line break.
